azure/scope/cluster_test.go

```diff
@@ -3438,3 +3439,246 @@ func TestVNetPeerings(t *testing.T) {
 		})
 	}
 }
+
+func TestPrivateEndpointSpecs(t *testing.T) {
+	tests := []struct {
+		name         string
+		clusterScope ClusterScope
+		want         []azure.ASOResourceSpecGetter[*asonetworkv1.PrivateEndpoint]
+	}{
+		{
+			name: "returns empty private endpoints list if no subnets are specified",
+			clusterScope: ClusterScope{
+				AzureCluster: &infrav1.AzureCluster{
+					Spec: infrav1.AzureClusterSpec{
+						NetworkSpec: infrav1.NetworkSpec{
+							Subnets: infrav1.Subnets{},
+						},
+					},
+				},
+				cache: &ClusterCache{},
+			},
+			want: make([]azure.ASOResourceSpecGetter[*asonetworkv1.PrivateEndpoint], 0),
+		},
```

Comment:
Should the `PrivateEndpointSpecs()` func return a `nil` array or an `empty` array? 
cc: @nojnhuh and @willie-yao